addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -53,6 +53,74 @@ def res_partner_enrich_and_create_company(self, partner_id):
 
         return response
 
+    @http.route('/mail_plugin/partner/enrich_and_update_company', type='json', auth='outlook', cors='*')
+    def res_partner_enrich_and_update_company(self, partner_id):
+        """
+        Enriches an existing company using IAP
+        """
+        partner = request.env['res.partner'].browse(partner_id)
+        if not partner.is_company:
+            return {'error': _('Contact must be a company')}
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            return {'error': _('Contact has no valid email')}
+
+        domain = tools.email_domain_extract(normalized_email)
+        iap_data = self._iap_enrich(domain)
+
+        if 'enrichment_info' in iap_data:  # means that an issue happened with the enrichment request
+            return {
+                'enrichment_info': iap_data['enrichment_info'],
+                'company': self._get_company_data(partner),
+            }
+
+        phone_numbers = iap_data.get('phone_numbers')
+
+        partner_values = {}
+
+        if not partner.phone and phone_numbers:
+            partner_values.update({'phone': phone_numbers[0]})
+
+        if not partner.iap_enrich_info:
+            partner_values.update({'iap_enrich_info': json.dumps(iap_data)})
+
+        if not partner.image_128:
```

Comment:
Why don't we check image_1920 by the way ?

Reply:
When a partner has an image, regardless of it's size, a image_128 will be created.

For this reason if a partner has no image_128 then they surely not have a image_1920.

Reply:
And so, why not direectly image 1920 :D ?

Reply:
The issue is that, if the image is smaller I think that image_1920 will be empty, and we will overwrite the contact's image.

Reply:
Image 128 is only a computed field (well it was the case a few months / years ago). Normally image_1920 is the main interface field. Could you check ?